jwst/pixel_replace/pixel_replace_step.py

```diff
@@ -81,9 +82,45 @@ def process(self, input):
                 'n_adjacent_cols': self.n_adjacent_cols,
             }
 
-            replacement = PixelReplacement(result, **pars)
-            replacement.replace()
-
-            self.record_step_status(replacement.output, 'pixel_replace', success=True)
-            return replacement.output
-
+            if isinstance(input_model, datamodels.ModelContainer):  # calspec3 case for IFU data
+                output_model = input_model.copy()
+                
+                # Setup output path naming if associations are involved.
+                asn_id = None
+                try:
+                    asn_id = self.input_model.meta.asn_table.asn_id
+                except (AttributeError, KeyError):
+                    pass
+                if asn_id is None:
+                    asn_id = self.search_attr('asn_id')
+                if asn_id is not None:
+                    _make_output_path = self.search_attr(
+                        '_make_output_path', parent_first=True
+                    )
+
+                    self._make_output_path = partial(
+                        _make_output_path,
+                        asn_id=asn_id
+                    )
+
+                for i, model in enumerate(input_model):
+                    if not isinstance(model, datamodels.IFUImageModel):
+                        self.log.error(f'Input is of type {str(type(input_model))} for which')
+                        self.log.error('pixel_replace does not have an algorithm.\n')
```

Comment:
Wait, I thought that pixel_replace was originally developed for non-IFU type data, such as MIRI LRS fixed-slit and NIRSpec fixed-slit spectra. So why is this saying there's no algorithm for non-IFU data?

Reply:
Sorry I had that in for testing. Will remove. I just can not get the output filename correct.

Reply:
This whole if-else block starting at line 109 could be greatly compacted by just doing:
```
if model.meta.model_type in ['MultiSlitModel', 'SlitModel', 'ImageModel', 'IFUImageModel', 'CubeModel']:
    self.log.debug('Input is a {model.meta.model_type}.')
else:
    self.log.error(f'Input is of type {model.meta.model_type} for which')
    self.log.error('pixel_replace does not have an algorithm.\n')
```
and perhaps keep the existing comments about what kinds of observing modes use each kind of model in just one block comment within the "if" clause.